tensorflow/lite/micro/tools/make/targets/apollo3_makefile.inc

```diff
@@ -67,13 +75,13 @@ LDFLAGS += \
   -fno-exceptions \
   -nostdlib --specs=nano.specs -t -lstdc++ -lc -lnosys -lm \
   -Wl,-T,$(TENSORFLOW_ROOT)$(APOLLO3_SDK)/boards/apollo3_evb/examples/hello_world/gcc_patched/apollo3evb.ld \
-  -Wl,-Map=$(TENSORFLOW_ROOT)$(MAKEFILE_DIR)/gen/$(TARGET).map,--cref
+  -Wl,-Map=$(TENSORFLOW_ROOT)$(MAKEFILE_DIR)/gen/$(BOARD)_$(TARGET).map,--cref
```

Comment:
if an incorrect board were to be specified, it would be nicer to have a clear error message about that, rather than some linker error about file not found.

I might suggest moving the if-else block for the board below to the top of the file so that you can provide an incorrect board error before anything else.